src/pages/edit-profile/EditProfile.tsx

```diff
@@ -103,6 +104,63 @@ const EditProfile = () => {
     }
   }, [loading, profileState, initialEditProfileState])
 
+  const changedProfileFields = useMemo<Partial<EditProfileState>>(() => {
+    if (!initialEditProfileState || !profileState) return {}
```

Comment:
I would recommend you to not return `{}` for non-changed profile fields. It's not obvious actualy.
Return `null` if profile fields are not changed, it will simplify logic for `isProfileChanged` variable calculation. You will have just `const isProfileChanged = Boolean(changedProfileFields)` without `useMemo`